gammapy/background/energy_offset_array.py

```diff
@@ -71,7 +95,11 @@ def _fill_one_event_list(self, events):
         ev_cube_hist, ev_cube_edges = np.histogramdd(ev_cube_array,
                                                      [self.energy.value,
                                                       self.offset.value])
-        return ev_cube_hist
+        energy_bin = np.diff(ev_cube_edges[0])
+        solid_angle_bin = np.pi * np.diff(ev_cube_edges[1] ** 2)
+        solid_angle_tab, energy_tab = np.meshgrid(solid_angle_bin, energy_bin)
+        bkg_rate = ev_cube_hist / (solid_angle_tab * energy_tab)
```

Comment:
Just to be sure that I understand:
We will have two class:
1)EnergyOffsetArray that just fill the events in an histogram
2)EnergyOffsetBackgroundModel that holds three arrays:
1. counts
2. livelime
3. rate
and where we add the I/O methods?

Right?

Le 9 févr. 2016 à 17:23, Lea Jouvin <ljouvin@apc.in2p3.fr> a écrit :

> ok!
> But in this case the write and read method should be for the class where we compute the background rate no?
> I will not write the I/O methods in the EnergyOffsetArray class?
> Le 9 févr. 2016 à 16:08, Christoph Deil <notifications@github.com> a écrit :
> 
>> In gammapy/background/energy_offset_array.py:
>> 
>> > @@ -71,7 +95,11 @@ def _fill_one_event_list(self, events):
>> >          ev_cube_hist, ev_cube_edges = np.histogramdd(ev_cube_array,
>> >                                                       [self.energy.value,
>> >                                                        self.offset.value])
>> > -        return ev_cube_hist
>> > +        energy_bin = np.diff(ev_cube_edges[0])
>> > +        solid_angle_bin = np.pi * np.diff(ev_cube_edges[1] ** 2)
>> > +        solid_angle_tab, energy_tab = np.meshgrid(solid_angle_bin, energy_bin)
>> > +        bkg_rate = ev_cube_hist / (solid_angle_tab * energy_tab)
>> My suggestion would be that you keep this pull request limited to the I/O and make a separate one with the new class and rate computation.
>> That way this one can be finished relatively quickly and review is easier.
>> 
>> I'll review the I/O methods later today ... for sure tests are needed.
>> Here's an example of a test that writes an object to FITS, reads it back, and checks that the content is OK:
>> https://github.com/gammapy/gammapy/blob/1fde3292a5f2f66da0be758249cb6f63fc478400/gammapy/scripts/tests/test_image_bin.py#L11
>> 
>> A similar test should be added here as well.
>> Make an example array, write, read back, assert something, e.g. an array entry value or first and last axis bin value.
>> 
>> Does this make sense, or should I describe in more detail how the I/O test should be implemented?
>> 
>> —
>> Reply to this email directly or view it on GitHub.
>> 
>